ansible/library/load_extra_dpu_config.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/python
+
+import paramiko
+import os
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.smartswitch_utils import smartswitch_hwsku_config
+
+DPU_HOST_IP_BASE = "169.254.200.{}"
+SRC_DPU_CONFIG_FILE = "/tmp/dpu_extra.json"
+DST_DPU_CONFIG_FILE = "/tmp/dpu_extra.json"
+DST_FULL_CONFIG_FILE = "/tmp/dpu_full.json"
+DEFAULT_CONFIG_FILE = "/etc/sonic/config_db.json"
+GEN_FULL_CONFIG_CMD = "jq -s '.[0] * .[1]' {} {} > {}".format(
+    DEFAULT_CONFIG_FILE, DST_DPU_CONFIG_FILE, DST_FULL_CONFIG_FILE)
+CONFIG_RELOAD_CMD = "sudo config reload {} -y".format(DST_FULL_CONFIG_FILE)
+CONFIG_SAVE_CMD = "sudo config save -y"
+
+
+class LoadExtraDpuConfigModule(object):
+    def __init__(self):
+        self.module = AnsibleModule(
+            argument_spec=dict(
+                hwsku=dict(type='str', required=True),
+                host_username=dict(type='str', required=False, default='admin'),
+                host_password=dict(type='str', required=False, default='password', no_log=True)
+            ),
+            supports_check_mode=False
+        )
+
+        self.hwsku = self.module.params['hwsku']
+        self.host_username = self.module.params['host_username']
+        self.host_password = self.module.params['host_password']
+
+        try:
+            self.hwsku_config = smartswitch_hwsku_config[self.hwsku]
+            self.dpu_num = self.hwsku_config.get('dpu_num', 0)
+
+            if self.dpu_num == 0:
+                self.module.fail_json(msg="No DPUs defined for hwsku: {}".format(self.hwsku))
+        except KeyError:
+            self.module.fail_json(msg="No DPU configuration found for hwsku: {}".format(self.hwsku))
+
+    def connect_to_dpu(self, dpu_ip):
+        """Establish an SSH connection to the DPU"""
+        try:
+            ssh = paramiko.SSHClient()
+            ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+            ssh.connect(dpu_ip, username=self.host_username, password=self.host_password, timeout=30)
+            return ssh
+        except Exception as e:
+            self.module.fail_json(msg="Failed to connect to DPU {}: {}".format(dpu_ip, str(e)))
+            return None
+
+    def transfer_to_dpu(self, ssh, dpu_ip):
+        """Transfer the configuration file to the DPU"""
```

Comment:
Is there a test case to verify the config upload to DPU correctly?

Reply:
I used catch exception to ccheck if sftp put fail

Reply:
No. But if it fail, there is a fail_json to catch that.